### DISCUSSION TITLE
Simple "Age/Founded" shortcode

### DISCUSSION THREAD

**Discussion by jollydodo** [thuliteio/doks#1258] [General]
This is not really directly related to Doks, but on the websites I've created it sometimes says things like "XX Years a household name in" or on my portfolio it for example mentions my current age. However, this is something that you might forget to update, so why not use the power of Hugo to automatically calculate the number of years since a certain date? I've messed around with different approaches and finally settled on the native "addDate" function which works quite well so I thought why not share my shortcode? (A little contribution to this great theme/website! However feel free to remove this if unwanted)

## How to: 

You can enter a default date in the shortcode (age.html) so that you can just refer to `{{< age >}} ` in your markdown or enter a parameter for a custom calculation like this: `{{< age "2022-06-20" >}} ` (Year, Month, Day). 

<img width="334" alt="Scherm­afbeelding 2024-07-22 om 19 31 16" src="https://github.com/user-attachments/assets/4e9c0548-c0f6-4b30-832e-c0c49c296a16">

## Installation: 

Simply create a new file called "age.html" in your shortcode folder. 

```Go
{{- /* Set defaults and get args. */}}
{{ $inputDate := index .Params 0 | default "2010-07-08" }} {{/* Year, Month, Day */}}

{{ $inputDate = time.Format "2006-1-2" $inputDate }} {{/* remove leading 0 if any */}}
{{ $dateComponents := split $inputDate "-" }}
{{ $year := sub 0 (index $dateComponents 0 | int) }}
{{ $month := sub 0 (index $dateComponents 1 | int) }}
{{ $day := sub 0 (index $dateComponents 2 | int) }}
{{ $age := now | time.AsTime }}
{{ $years := $age.AddDate (int $year) (int $month) (int $day) | time.Format "06" }} {{/* Turn int64 to int */}}

{{- /* Display age in years */}}
{{ int $years }}`
```

## Remarks: 

There is no rounding. So 1 year and 11 months will return 1.

**Comment by h-enk** [comment]
Nice one — thanks for sharing!
